packages/server/lib/controllers/sync.controller.ts

```diff
@@ -687,6 +694,124 @@ class SyncController {
             next(e);
         }
     }
+
+    /**
+     * PUT /sync/frequency
+     *
+     * Allow users to change the default frequency value of a sync without losing the value.
+     * The system will store the value inside `_nango_syncs.frequency` and update the relevant schedules.
+     */
+    public async updateFrequencyForConnection(req: Request, res: Response, next: NextFunction) {
+        try {
+            const { sync_name, provider_config_key, connection_id, frequency } = req.body;
+
+            // Payload validation
+            if (!provider_config_key || typeof provider_config_key !== 'string') {
+                res.status(400).send({ message: 'provider_config_key must be a string' });
+                return;
+            }
+            if (!sync_name || typeof sync_name !== 'string') {
+                res.status(400).send({ message: 'sync_name must be a string' });
+                return;
+            }
+            if (!connection_id || typeof connection_id !== 'string') {
+                res.status(400).send({ message: 'connection_id must be a string' });
+                return;
+            }
+            if (typeof frequency !== 'string' && frequency !== null) {
+                res.status(400).send({ message: 'frequency must be a string or null' });
+                return;
+            }
+
+            // Validate new frequency format
+            let newFrequency: string | undefined;
+            if (frequency) {
+                const { error, response } = getInterval(frequency, new Date());
+                if (error || !response) {
+                    res.status(400).send({ message: 'frequency must have a valid format (https://github.com/vercel/ms)' });
+                    return;
+                }
+                newFrequency = response.interval;
+            }
+
+            // Get implicit env
+            const getEnv = await getEnvironmentAndAccountId(res, req);
+            if (!getEnv.success || getEnv.response === null) {
+                errorManager.errResFromNangoErr(res, getEnv.error);
+                return;
+            }
+            const envId = getEnv.response.environmentId;
+
+            // Fetch connection
+            const getConnection = await connectionService.getConnection(connection_id, provider_config_key, envId);
+            if (!getConnection.response || getConnection.error) {
+                res.status(400).send({ message: 'Invalid connection_id' });
+                return;
+            }
+            const connection = getConnection.response;
+
+            // Verify sync
+            const syncs = await findSyncByConnections([Number(connection.id)], sync_name);
+            if (syncs.length <= 0) {
+                res.status(400).send({ message: 'Invalid sync_name' });
+                return;
+            }
+            const syncId = syncs[0]!.id!;
+
+            // Verify schedule
```

Comment:
This logic until 772 seems to be a repeat of the logic in `updateSyncScheduleFrequency`. Can that be avoided?

Reply:
Ah yes indeed didn't saw there was already this check inside 👌🏻